http/codegen/openapi/json_schema.go

```diff
@@ -100,16 +105,63 @@ const (
 )
 
 // SchemaRef is the JSON Hyper-schema standard href.
-const SchemaRef = "http://json-schema.org/draft-04/hyper-schema"
+// Updated to JSON Schema 2020-12 for OpenAPI 3.1 compatibility
+const SchemaRef = "https://json-schema.org/draft/2020-12/schema"
 
 var (
 	// Definitions contains the generated JSON schema definitions
 	Definitions map[string]*Schema
+	
+	// GenerateForOpenAPIv2 indicates whether we're generating for OpenAPI v2 (Swagger)
+	// When true, schemas will use v2 compatible format (single example, string type)
+	// When false, schemas will use v3.1 format (examples array, type arrays for nullable)
+	GenerateForOpenAPIv2 bool
```

Comment:
It would be a lot cleaner to bite the bullet and create a separate struct. Maybe the v2 and v3.1 structs can embed a common struct for most things which would implement the common behavior. Having this global variable trigger different behavior isn't great.